docs/development/component-checklist.md

```diff
@@ -125,6 +125,12 @@ This document provides a checklist for them that you can walk through.
 
    <sub>[*] It is required because if a component deployed in the Shoot cluster does not specify a Seccomp profile and cannot run with the `RuntimeDefault` Seccomp profile, then enabling the `.spec.kubernetes.kubelet.seccompDefault` field in the Shoot spec would break the corresponding component.</sub>
 
+9. **Use strict mode when decoding data** ([example](https://github.com/gardener/gardener/blob/485c25124ea536e4610c627109017dd34d434921/pkg/provider-local/controller/worker/actuator.go#L52))
```

Comment:
Can we give example of strict validation for a decoder used in admission component? I think there it makes more sense and the example is much more obvious. The admission rejects introducing an unknown field. When the example decoder is in actuator/controller, there the unknown field is already part of the resource/API. And the best thing the controller can do is to do fail the reconciliation and not reconcile the resource.